### PR TITLE
fix: 🐛 issue#276 fix. Added eip712 domain validations

### PR DESCRIPTION
**Pull request by oleggrib**
✅ Closes: #276

### CODE REVIEW

File: src/main/javascript/crypto/src/Authenticator.ts [hunk 1]
```diff
@@ -197,9 +196,13 @@ export class Authenticator {
 
         let pok = crypto.computeAttestationProof(secret, nonce);
         let attRequest = AttestationRequest.fromData(crypto.getType(type), pok);
-        let attest = new Eip712AttestationRequest(userKey);
-        await attest.addData(attestorDomain, 20*1000, receiverId, attRequest);
-        let attestJson = attest.getJsonEncoding();
+        let attestationRequest = new Eip712AttestationRequest(userKey);
+        await attestationRequest.addData(attestorDomain, undefined, receiverId, attRequest);
```
Comment (jot2re) [inline src/main/javascript/crypto/src/Authenticator.ts:200]: Why `undefined` validity? How exactly is that encoded? Shouldn't it be default validity?

**Review by jot2re — CHANGES_REQUESTED** [review]
Thanks for the update! I think there might still be a couple of checks that should be added. 
In particular I think that some updates are also needed to the validation in Eip712AttestationRequestWithUsage and Eip712AttesationUsage in the sense that they should perform the same domain validation as Eip712AttesationRequest.

File: src/main/javascript/crypto/src/Authenticator.ts (hunk 1 above)
Comment (oleggrib) [reply src/main/javascript/crypto/src/Authenticator.ts:200]: when function receive "undefined" then it use default value

File: src/main/javascript/crypto/src/libs/Eip712Validator.ts [hunk 2]
```diff
@@ -42,56 +46,61 @@ export class Eip712Validator {
         return this.domain;
     }
 
-    validateRequest(jsonInput: string) {
-        try {
-            let authenticationData = JSON.parse(jsonInput);
+    // we dont use it in Java code
```
[the remark this answers is not in the document]
Comment (oleggrib) [reply src/main/javascript/crypto/src/libs/Eip712Validator.ts]: removed

**Comment by jot2re** [comment]
The issue of missing domain validation turned out to also be present in attestation.jar. I handled this in [this PR](https://github.com/TokenScript/attestation/pull/285)

**Comment by jot2re** [comment]
I also implemented the JS fix in [this commit ](https://github.com/TokenScript/attestation/pull/281/commits/8c5da8a2a5786ee36602c5b18ab359f97f5e9174). So I am happy with merging if you are @oleggrib

**Review by jot2re — APPROVED** [review]

**Pull request merged**